grakn-engine/src/main/java/ai/grakn/engine/cache/EngineCacheDistributed.java

```diff
@@ -220,6 +221,20 @@ public void clearAllJobs(String keyspace) {
         clearAllJobs(RESOURCE_JOB, keyspace);
         clearAllJobs(CASTING_JOB, keyspace);
     }
+
+    @Override
+    public long getLastTimeJobAdded() {
+        Optional<Object> value = getObjectFromZookeeper(ENGINE_CACHE_UPDATE_TIME);
+        if(value.isPresent()){
+            return (long) value.get();
+        }
+        return 0L;
```

Comment:
Maybe:
`return getObjectFromZookeeper(ENGINE_CACHE_UPDATE_TIME).orElse(0L);`
but you might need a nasty cast...